x-pack/plugins/osquery/public/editor/osquery_highlight_rules.ts

```diff
@@ -102,84 +100,141 @@ const dataTypes = [
+];
 
-// This is gross, but the types exported by brace are lagging and incorrect: https://github.com/thlorenz/brace/issues/182
-(ace as unknown as AceInterface).define(
-  'ace/mode/osquery_highlight_rules',
-  ['require', 'exports', 'ace/mode/sql_highlight_rules'],
-  // eslint-disable-next-line prefer-arrow-callback
-  function (acequire, exports) {
-    'use strict';
+interface Range {
+  startLineNumber: number;
+  endLineNumber: number;
+  startColumn: number;
+  endColumn: number;
+}
 
-    const SqlHighlightRules = acequire('./sql_highlight_rules').SqlHighlightRules;
+interface IDisposable {
+  dispose: () => void;
+}
 
-    class OsqueryHighlightRules extends SqlHighlightRules {
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      constructor(...args: any) {
-        super(...args);
-        const keywordMapper = this.createKeywordMapper(
-          {
-            'osquery-token': osqueryTables,
-            'support.function': builtinFunctions,
-            keyword: keywords,
-            'constant.language': builtinConstants,
-            'storage.type': dataTypes,
-          },
-          'identifier',
-          true
-        );
+const theme = {
+  base: 'vs' as const,
+  inherit: false,
+  rules: [
+    { token: 'osquery-token' },
+    { token: 'support.function', foreground: '4271AE' },
+    { token: 'keyword', foreground: '8959A8' },
+    { token: 'storage.type', foreground: '8959A8' },
+    { token: 'constant.language', foreground: 'F5871F' },
+    { token: 'comment', foreground: '8E908C' },
+    { token: 'string', foreground: '718C00' },
+    { token: 'constant.numeric', foreground: 'F5871F' },
+    { token: 'keyword.operator', foreground: '3E999F' },
+  ],
+  colors: {},
+};
 
-        this.$rules = {
-          start: [
-            {
-              token: 'comment',
-              regex: '--.*$',
-            },
-            {
-              token: 'comment',
-              start: '/\\*',
-              end: '\\*/',
-            },
-            {
-              token: 'string', // " string
-              regex: '".*?"',
-            },
-            {
-              token: 'string', // ' string
-              regex: "'.*?'",
-            },
-            {
-              token: 'constant.numeric', // float
-              regex: '[+-]?\\d+(?:(?:\\.\\d*)?(?:[eE][+-]?\\d+)?)?\\b',
-            },
-            {
-              token: keywordMapper,
-              regex: '[a-zA-Z_$][a-zA-Z0-9_$]*\\b',
-            },
-            {
-              token: 'keyword.operator',
-              regex: '\\+|\\-|\\/|\\/\\/|%|<@>|@>|<@|&|\\^|~|<|>|<=|=>|==|!=|<>|=',
-            },
-            {
-              token: 'paren.lparen',
-              regex: '[\\(]',
-            },
-            {
-              token: 'paren.rparen',
-              regex: '[\\)]',
-            },
-            {
-              token: 'text',
-              regex: '\\s+',
-            },
+export const initializeOsqueryEditor = () => {
```

Comment:
How do you feel about moving this to the `kibana_react` plugin? I think it will make it easier next time we need to have an editor with osquery syntax rules and there seems to be a section already defined for adding languages [here](https://github.com/elastic/kibana/blob/main/src/plugins/kibana_react/public/code_editor/register_languages.ts).

Reply:
Thanks for reviewing the PR 👍  As discussed offline, I'll leave it as it is because it's only gonna be used with Osquery editor, it's rather not a language itself.